src/components/Match/MatchStory.jsx

```diff
@@ -326,41 +335,70 @@ class LanesEvent extends StoryEvent {
   }
 }
 
+// returnes a formatted template when given a TowerEvent or BarracksEvent
+const formatBuilding = (event) => {
+  let template = strings.story_building_destroy;
+  if (event.player) {
+    template = event.player.isRadiant === event.team ? strings.story_building_deny_player : strings.story_building_destroy_player;
```

Comment:
for events, `team` is a boolean where radiant is true, and dire is false. It's different from how they are passed in through the objectives array.